scripts/updateUserCountry.js

```diff
@@ -0,0 +1,72 @@
+// Script to update slugified country values for user.country and user.geo.countryName
+
+require('coffee-script');
+require('coffee-script/register');
+global._ = require('lodash')
+_.str = require('underscore.string')
+database = require('../server/commons/database')
+User = require('../server/models/User')
+co = require('co')
+mongoose = require('mongoose')
+
+database.connect()
+var count = 0
+var query = {"_id": {$gt: mongoose.Types.ObjectId('000000000000000000000000')}, 
+            $or: [
+                {"geo": {$exists: true}},
+                {"country": {$exists: true}}
+            ]
+        }
+
+updateUser = co.wrap(function*(user) {
+    geo = user.get("geo")
+    country = user.get("country")
+    if (geo && geo.countryName){
+        countryNameSlug = _.str.slugify(geo.countryName)
+        user.set('geo.countryName', countryNameSlug)
+        if (!country){
+            user.set("country", countryNameSlug)
+        }
+    }
+    if (country){
+        user.set("country", _.str.slugify(country))
+    }
+    database.validateDoc(user)
+    try{
+        res = yield user.save()
+        count += 1
+        console.log(new Date().toISOString(), "Updated user id: ", res._id, " with timestamp: ", res._id.getTimestamp())
+    }
+    catch(e){
+        console.log("Error in save for :", user)
+    }
+})
+
+co(function*() {
+    while (true){
+        console.log("Starting...")
+        var users = yield User.find(query).limit(100).sort({_id:1})  // change limit as required
```

Comment:
Do we need to sort before limit?  I would be leery of limiting to a random 100 and then sorting those, unless this not how MongoDB is going to treat this chaining.

Reply:
Yeah, good point. But it somehow worked fine while updating geo location for existing users somehow. 
Anyways, I am checking how does mongoose treat this chaining, and it would probably be safer to first sort and then limit.

Reply:
I read that the order in which we write sort() and limit() does not matter - mongo always perform them in the order of sort first and then limit (https://docs.mongodb.com/manual/reference/method/db.collection.find/#combine-cursor-methods)
But I am still changing this because it seems better/cleaner to write in the correct order.